tools/wakebox/wakebox.cpp

```diff
@@ -259,25 +270,25 @@ int main(int argc, char *argv[]) {
     return 1;
   }
 
+  std::vector<mount_op> bind_ops;
+  for (unsigned int i = 0; i < arg(options, "bind")->count; i++) {
+    const std::string s = binds[i];
+    std::string source = s.substr(0, s.find(":"));
+    std::string destination = s.substr(s.find(":") + 1);
+    if (source.empty() || destination.empty() || s.find(":") == std::string::npos) {
+      std::cerr << "Invalid bind: " << s << std::endl;
+      return 1;
+    }
+    bind_ops.push_back({"bind", source, destination});
+  }
+
```

Comment:
Looks like the macOS builds are failing. When I was reading the code, I was actually kind of curious why this wasn't broken on macOS, given that the `--bind` option is declared under an `#ifdef __linux__`, but I guess the answer is that it _is_ broken on macOS but CI isn't exercising that case because we only try to read the `--bind` option under wakebox's interactive mode.

Should I put everything in this `for` loop in an `#ifdef __linux__` but leave the `bind_ops` outside so that on macOS it just becomes an empty vector?

Reply:
As we don't support any mount options on MacOS other than placement of the fuse-waked fuse mount, I think it's reasonable to have `bind_ops` as linux-only (or empty for non-linux)

Reply:
Mounting is typically an ordered activity, so I'm wondering if users will care about the ordering here, say if you write
```
wakebox --bind $dir1:/dest1 -p params.json --bind $dir2:/dest2
```
Perhaps the user may want to use `--bind $dir1:/dest1` to modify what's happening before `params.json`.
Though if they're replacing the rootfs I guess it may be less of a use-case to move things around beforehand

Reply:
> Mounting is typically an ordered activity, so I'm wondering if users will care about the ordering here, say if you write

I was wondering about this myself, but I wasn't sure if that was something that we could easily handle with the current option parsing library, since I think, by default, all the `--bind` values get grouped into a single array.

Is this something I should look into? I think we may need to change our option parsing approach to be more of an imperative builder pattern rather than the current declarative approach if we want to keep track of the relative order of different options.

Reply:
We could just put something in the `--help` saying that the bind options always happen after(?) the params file is read

Reply:
Oh you already have that in the --help